src/proteus/dswx_hls.py

```diff
@@ -3432,6 +3434,58 @@ def _populate_dswx_metadata_datasets(dswx_metadata_dict,
         dswx_metadata_dict['SHORELINE_SOURCE'] = '(not provided)'
 
 
+def _populate_dswx_metadata_processing_parameters(
+        dswx_metadata_dict,
+        shadow_masking_algorithm,
+        min_slope_angle,
+        max_sun_local_inc_angle,
+        mask_adjacent_to_cloud_mode,
+        forest_mask_landcover_classes,
+        ocean_masking_shoreline_distance_km):
+    """Populate metadata dictionary with processing parameters
+
+       Parameters
+       ----------
+       dswx_metadata_dict : collections.OrderedDict
+              Metadata dictionary
+       shadow_masking_algorithm: str
+              Shadow masking algorithm
+       min_slope_angle: float
+              Minimum slope angle in degrees for terrain masking
+       max_sun_local_inc_angle: float
+              Maximum sun local-incidence angle in degrees for terrain masking
+       mask_adjacent_to_cloud_mode: bool
+              Define how areas adjacent to cloud/cloud-shadow should be handled
+       forest_mask_landcover_classes: list
+              Copernicus CGLS Land Cover 100m forest classes to mask out from
+              the WTR-2 and WTR layer due to dark reflectance that is usually
+              misinterpreted as water.
+       ocean_masking_shoreline_distance_km: float
+              Ocean masking distance from shoreline in km
+    """
+
+    # shadow masking algorithm and parameters
+    dswx_metadata_dict['SHADOW_MASKING_ALGORITHM'] = shadow_masking_algorithm
+    if shadow_masking_algorithm == 'sun_local_inc_angle':
+        dswx_metadata_dict['MIN_SLOPE_ANGLE'] = min_slope_angle
+        dswx_metadata_dict['MAX_SUN_LOCAL_INC_ANGLE'] = max_sun_local_inc_angle
+    else:
+        dswx_metadata_dict['MIN_SLOPE_ANGLE'] = '-'
+        dswx_metadata_dict['MAX_SUN_LOCAL_INC_ANGLE'] = '-'
+
+    # mask adjacent to cloud/cloud shadow mode
+    dswx_metadata_dict['MASK_ADJACENT_TO_CLOUD_MODE'] = \
+        mask_adjacent_to_cloud_mode
+
+    # Copernicus class for forest masking
+    dswx_metadata_dict['FOREST_MASK_LANDCOVER_CLASSES'] = \
+        ', '.join(forest_mask_landcover_classes)
```

Comment:
I left this comment elsewhere, but shouldn't class 201 (evergreen) also be included in this list for the metadata?